src/virtual-machine.js

```diff
@@ -419,16 +420,14 @@ class VirtualMachine extends EventEmitter {
 
     createSatelliteVariables (data) {
         const stage = this.runtime.getTargetForStage();
-        let singleSat = stage.lookupVariableByNameAndType(`${data}`, '');
         let allSats = stage.lookupVariableByNameAndType('All_Satellites', 'list');
+        let singleSat = stage.lookupVariableByNameAndType(`${data}`, '');
         if (!allSats) {
             allSats = this.workspace.createVariable(`All_Satellites`, 'list', false, false);
+        }
+        if (!singleSat) {
```

Comment:
This is the key change to allow all the individual sat IDs to be created as variables